debug_module.tex

```diff
@@ -177,24 +177,24 @@ \section{Hart States}
 \section{Run Control} \label{runcontrol}
 
 For every hart, the Debug Module contains 4 conceptual bits of state: halt
-request, resume request, halt-on-reset request,  and hart reset.
+request, resume ack, halt-on-reset request,  and hart reset.
 (The hart reset and halt-on-reset request bits are optional.)
 These bits all reset to 0. A debugger can write them for the currently selected
-harts through \Fhaltreq, \Fresumereq, \Fsetresethaltreq/\Fclrresethaltreq and
-\Fhartreset in \Rdmcontrol, only one of which may be high in any single write.
-In addition the DM receives halted, running, and resume ack signals from each
-hart.
+harts through \Fhaltreq, \Fsetresethaltreq/\Fclrresethaltreq and
+\Fhartreset in \Rdmcontrol.
+In addition the DM receives halted and running signals from each hart.
 
-When a running hart receives a halt request, it responds by halting,
+When a running hart sees its halt request bit high, it responds by halting,
 deasserting its running signal, and asserting its halted signal. The halted
 signals of all selected harts are reflected in the \Fallhalted and \Fanyhalted
-bits. \Fhaltreq is ignored by halted harts.
+bits. Halted harts ignore their halt request bit.
 
 When a halted hart receives a resume request, it responds by resuming, clearing
-its halted signal, and asserting its running signal and resume ack signals. The
-resume ack signal is lowered when the resume request is deasserted.  These
+its halted signal, and asserting its running signal. The DM
+reacts to a hart's running signal going high by setting resume ack for that
```

Comment:
I don't know why we need to be so prescriptive about what the debug module does and what the hart does. I would rewrite this to say:

When the DM receives a resume request for a hart, that hart's resumeack bit is cleared and it's resume req bit is set. The hart responds by resuming, clearing its halted signal, and asserting its running signal. In this process the resumeack bit is cleared, either by the hart or the DM.

Reply:
The DM clearing resume ack is already in the register definition, but we could add it here as well.

There is no resume request bit. There might be, but it's not state that has any reason to persist, beyond for implementation details.

How about:
When a debugger writes 1 to \Fresumereq, each selected hart's resume ack bit is cleared, and each selected, halted hart is sent a resume request. Harts respond by resuming, clearing their halted signal, and asserting their running signal. At the end of this process the resume ack bit is cleared.

Reply:
Yes, that makes more sense to me.

Reply:
Changed.